extide/gradle/src/org/netbeans/modules/gradle/spi/execute/JavaRuntimeManager.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.gradle.spi.execute;
+
+import java.io.File;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import javax.swing.event.ChangeListener;
+import org.netbeans.api.annotations.common.NonNull;
+
+/**
+ *
+ * @author Laszlo Kishalmi
+ */
+public interface JavaRuntimeManager {
+
+    public static String DEFAULT_RUNTIME_ID = "default_platform"; //NOI18N
```

Comment:
Question: how to handle if this ID clashes with some real JavaPlatform's ID ?

Reply:
Well, it is planned to be "clashing". That's the ID actually used by the IDE to mark the default JavaPlatform. The ID-s are never really made into the JavaPlatform API, though used a lot of places. I'm a bit unsure if the ID-s need to be part of this API, as it could be implemented without exposing them this much, though it made some use cases more simple.